dvc/repo/experiments/utils.py

```diff
@@ -33,7 +33,7 @@ def exp_refs_by_name(
 ) -> Generator["ExpRefInfo", None, None]:
     """Iterate over all experiment refs matching the specified name."""
     for ref_info in exp_refs(scm):
-        if ref_info.name == name:
+        if ref_info.name == name or str(ref_info) == name:
```

Comment:
If `name` is an absolute ref (meaning it starts with `refs/`) we don't have to do anything to find the matching ref, we can just return the absolute ref.

Doing the string search/loop iteration is potentially slow, which was why we had the `if name.startswith("refs/")` shortcut checks in the calling functions. The bug in the shortcut implementation was just that we were returning the absolute refs as strings and not as `ExpRefInfo`s (https://github.com/iterative/dvc/issues/6421)

Basically we just want something like:
```py
if name.startswith("refs/"):
    try:
        yield ExpRefInfo.from_ref(name)
    except InvalidExpRefError:
        pass
    return
```

Reply:
Yes, but I guess for the remote refs, this would not cause any performance issue? Because we didn't initialize the remote repo previously?

Reply:
Maybe we should rename it to `exp_ref_info_by_ref`?

Reply:
If the user is providing an absolute `refs/...` string, we can just assume that it exists, either locally or in the remote, and error out later when we try to do whatever we need to with that ref if it turns out to not exist. We don't need to iterate over anything in this case.

It doesn't matter  `ExpRefInfo.from_ref("refs/...")` doesn't matter if it's a local or remote ref, the string is the same either way.

Reply:
This change can be removed, as discussed we should be doing the check and using `from_ref` elsewhere here. This function is specifically for searching for experiments by the `name` segment of an experiment ref

Reply:
@karajan1001